.github/workflows/build.yml

```diff
@@ -166,7 +166,7 @@ jobs:
   build-macos-x86_64:
     needs: [generate-license]
     name: Mac x86_64
-    runs-on: macos-latest
+    runs-on: macos-15-intel
```

Comment:
I'm not sure, why would we test it on intel chipset, arm (latest) makes more sense to me

Reply:
making this change because this was [originally using `macos-13`](https://github.com/apache/datafusion-ballista/pull/1324/files#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721L169)
macos-15-intel is the direct replacement

Reply:
the name of the step "Mac x86_64" suggest that we should use an intel based processor. 

macos-latest is arm64
https://docs.github.com/en/actions/reference/runners/github-hosted-runners#standard-github-hosted-runners-for-public-repositories

Reply:
I believe intel architecture for Mac is on its way out, let's keep arm, and just rename action, Wdyt?